app/main.py

```diff
@@ -1,8 +1,21 @@
 class Person:
-    # write your code here
-    pass
+    people = {}
+
+    def __init__(self, name: str, age: int) -> None:
+        self.name = name
+        self.age = age
+        Person.people[self.name] = self
 
 
 def create_person_list(people: list) -> list:
-    # write your code here
-    pass
+    persons = []
+    for citizen in people:
+        person = Person(citizen["name"], citizen["age"])
+        if "wife" in citizen and citizen["wife"] is not None:
+            for human in people:
+                if citizen["wife"] == human["name"]:
+                    person_wife = Person(human["name"], human["age"])
```

Comment:
I believe here you potentially create duplications

Try to implement a solution with two consecutive loops